scripts/testing/create_override_csv.py

```diff
@@ -0,0 +1,132 @@
+import os
+import sys
+
+
+def get_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "-k",
+        "--key", nargs="+", default=[],
+        help="list of keys")
+    parser.add_argument(
+        "-d",
+        "--dir", nargs="+", default=['.'],
+        help="list of dirs")
+    parser.add_argument(
+        "-o",
+        "--output", default='override_combined.csv',
+        type=str,
+        help="output file (default override_combined.csv)")
+
+    parser.add_argument(
+        "-v",
+        "--verbose",
+        action="count",
+        default=0,
+        help="Make the output more verbose, -v shows csv files, -vv also cats the final output")
+    args = parser.parse_args()
+    return args
+
+
+def make_combos(keys, base, file_list, level):
+    """ Create combinations of keys for a particular level upwards.
+
+    Keyword arguments:
+        keys      -- List of keys to combine, suitable for the base
+        base      -- base filename to extend
+        file_list -- List of lists of filenames, one list for each level
+        level     -- Current level
+
+    We want to create combinations of keys based on maintaining the order passed
+    in and combining with any keys which are beyond them in the list. Also we
+    want to make combinations of different numbers of keys each at a distinct
+    level. e.g. if we have keys one,two,three, we'd have keys L 1: one, two,
+    three, L 2: one_two, one_three, two_three L 3: one_two_three. make_combos
+    uses a list of keys and a base for the filename as well as file_list which
+    is a list of lists. It also takes a level which tells it which sub-list to
+    add to. It then calls make_combos recursively to work up the levels but
+    without the first key in it's list which it uses to extend the base.
+    """
+    for i, _ in enumerate(keys):
+        new_combo = base + "_" + keys[i]
+        file_list[level].append(new_combo + ".csv")
+        # Combine to the next level up but starting from the next key
+        # this avoids recreating combos and keeping the order
+        make_combos(keys[i + 1:], new_combo, file_list, level + 1)
+
+
+def concat_files(output_file, dirs, file_list, verbose):
+    """ Concat any files found in the file_list in any of the directories
+
+        Keyword arguments:
+        output_file      -- Output csv file
+        dirs             -- List of directories to try for each file in file_list
+        file_list        -- List of lists of filenames, one list for each level
+        verbose          -- Verbosity level (0,1 or 2)
+    Search through the file list and directories to find files in the file_list.
+    Concat them if found onto the output_file.
+    """
+    for f in file_list:
+        for d in dirs:
+            filename = os.path.join(d, f)
+            if os.path.exists(filename):
+                if verbose > 0:
+                    print("Found: ", filename)
+                with open(filename, "r") as fp:
```

Comment:
Apologies if the syntax isn't quite right yet but...
```suggestion
            try:
                fp = open(filename, "r")
            except FileNotFoundError:
                continue
            with fp:
                if verbose > 0:
                    print("Found: ", filename)
```